NetworkPkg/HttpBootDxe/HttpBootImpl.c

```diff
@@ -370,13 +374,38 @@ HttpBootGetBootFileCaller (
         //
         // Load the boot file into Buffer
         //
-        Status = HttpBootGetBootFile (
-                   Private,
-                   FALSE,
-                   BufferSize,
-                   Buffer,
-                   ImageType
-                   );
+        for (Retries = 0; Retries < PcdGet32 (PcdMaxHttpResumeRetries); Retries++) {
+          Status = HttpBootGetBootFile (
+                     Private,
+                     FALSE,
+                     BufferSize,
+                     Buffer,
+                     ImageType
+                     );
+          if (!EFI_ERROR (Status) || (Status != EFI_NOT_READY)) {
+            break;
+          }
+
+          //
+          // HttpBootGetBootFile returned EFI_NOT_READY, we may attempt to resume
+          // the interrupted download.
+          //
+
+          Private->HttpCreated = FALSE;
+          HttpIoDestroyIo (&Private->HttpIo);
+          Status = HttpBootCreateHttpIo (Private);
+          if (EFI_ERROR (Status)) {
+            break;
+          }
+
+          DEBUG ((DEBUG_WARN | DEBUG_INFO, "HttpBootLoadFile: Download interrupted, will try to resume the operation.\n"));
+          gBS->Stall (1000 * 1000 * PcdGet32 (PcdHttpDelayBetweenResumeRetries));
+        }
+
+        if (Status == EFI_NOT_READY) {
+          Status = EFI_TIMEOUT;

```

Comment:
Can you add a DEBUG_ERROR message here saying we've timed out and hence failed downloading the file? It'll help with debugging failures.

Reply:
Sure.

Reply:
In fact there is an error, the Status will never be equal EFI_NOT_READY outside the for loop. I will change the for loop, putting the for conditition inside the loop.

Reply:
I will wait to push the changes. There is a lot of pull requests being integrated to master branch right now, everytime I push to this pull request, it gets out-of-date in a matter of minutes.

Reply:
Its ok if its out of date. As long as there's no merge conflict, you're good.

Reply:
Ok, I just pushed my branch.